File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
values = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

def label(colors):

    sum = ""
    zeros = ""      
       
    for color in colors[:2]:
        sum += str(values.index(color))
    
    zeros = "0" * values.index(colors[2])    
   
    total =  sum+zeros    
    result = int(total)

    prefixes = [" ohms", " kiloohms", " megaohms", " gigaohms"]
    prefix_index = 0
    while result >= 1000:
        result = result // 1000
        prefix_index += 1

    
    label = f"{result}{prefixes[prefix_index]}"
   
    return label
    

File: solutions/java/resistor-color-trio/1/src/main/java/ResistorColorTrio.java
import java.util.Arrays;
import java.util.List;

class ResistorColorTrio {

    private static final List<String> COLORS = Arrays.asList(
        "black", "brown", "red", "orange", "yellow",
        "green", "blue", "violet", "grey", "white"
    );

    String label(String[] colors) {
        int firstDigit = COLORS.indexOf(colors[0]);
        int secondDigit = COLORS.indexOf(colors[1]);
        int exponent = COLORS.indexOf(colors[2]);

        
        long mainValue = firstDigit * 10 + secondDigit;
        long ohms = mainValue * (long) Math.pow(10, exponent);

        if (ohms >= 1_000_000_000) {
            return (ohms / 1_000_000_000) + " gigaohms";
        } else if (ohms >= 1_000_000) {
            return (ohms / 1_000_000) + " megaohms";
        } else if (ohms >= 1_000) {
            return (ohms / 1_000) + " kiloohms";
        } else {
            return ohms + " ohms";
        }
    }
}